pymodbus/client/tcp.py

```diff
@@ -54,24 +57,45 @@ async def run():
 
     socket: socket.socket | None
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         host: str,
-        port: int = 502,
         framer: FramerType = FramerType.SOCKET,
+        port: int = 502,
```

Comment:
For reference: this breaks the interface for those who used port as a positional argument

Reply:
Yes, as noted in API changes.